platform/pulp/platform/models/repository.py

```diff
@@ -213,44 +225,20 @@ class Importer(Plugin):
 
     scratchpad = fields.GenericRelation(Scratchpad)
 
-    class Meta:
-        abstract = True
-
-
-class RepositoryImporter(Importer):
-    """
-    A content importer that is associated with a repository.
-
-    Fields:
-
-    Relations:
-
-    :cvar repository: The associated repository.
-    :type repository: models.ForeignKey
-    """
-
     repository = models.ForeignKey(
-        Repository, related_name='importers', on_delete=models.CASCADE)
-
-    class Meta:
-        unique_together = ('repository', 'name')
-
-    def natural_key(self):
-        """
-        Get the model's natural key.
-
-        :return: The model's natural key.
-        :rtype: tuple
-        """
-        return (self.repository.id, self.name)
+        Repository, blank=True, null=True, related_name='importers', on_delete=models.CASCADE)
 
 
-class Distributor(Plugin):
+class Publisher(ContentAdaptor):
     """
-    An abstract content distributor.
+    A content publisher.
 
     Fields:
 
+    :cvar auto_publish: Indicates that the adaptor may publish automatically
```

Comment:
When I read "may" I feel uncertain about what this field really means, since it's a Boolean field. I know this is very nitpicky, but maybe "will" is better here?

![Do or do not. There is no "may"](https://i.ytimg.com/vi/BQ4yd2W50No/hqdefault.jpg)